src/test/java/com/snowplowanalytics/snowplow/tracker/emitter/CollectorCookieJarTest.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2014-2022 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.tracker.emitter;
+
+import com.snowplowanalytics.snowplow.tracker.DevicePlatform;
+import com.snowplowanalytics.snowplow.tracker.Tracker;
+import com.snowplowanalytics.snowplow.tracker.events.PageView;
+import com.snowplowanalytics.snowplow.tracker.events.ScreenView;
+import com.snowplowanalytics.snowplow.tracker.http.CollectorCookieJar;
+import com.snowplowanalytics.snowplow.tracker.payload.TrackerPayload;
+import okhttp3.Cookie;
+import okhttp3.HttpUrl;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+public class CollectorCookieJarTest {
+    String domain1 = "http://snowplow.test.url.com";
+    String domain2 = "http://other.test.url.com";
+    Cookie cookie1;
+    CollectorCookieJar cookieJar;
+    ArrayList<Cookie> requestCookies;
+
+    @Before
+    public void setUp() {
+        cookie1 = new Cookie.Builder()
+                .name("sp")
+                .value("xxx")
+                .domain("snowplow.test.url.com")
+                .build();
+        cookieJar = new CollectorCookieJar();
+        requestCookies = new ArrayList<>();
```

Comment:
Really tiny nitpick and probably a subjective preference, but there is not much need to make this list an instance variable. I would prefer to initialize it in every test (maybe as `Arrays.asList(cookie1)`). Making it an instance variable suggests that there is some shared state assigned to it in the `setUp()` and makes it slightly more difficult to read the tests as one has to check back for the initialization in the `setUp()`.